tests/x509/test_x509.py

```diff
@@ -1232,6 +1232,45 @@ def test_load_rsa_certificate_request(self, path, loader_func, backend):
         assert isinstance(extensions, x509.Extensions)
         assert list(extensions) == []
 
+    def test_get_attribute_for_oid(self, backend):
+        request = _load_cert(
+            os.path.join(
+                "x509", "requests", "challenge.pem"
+            ), x509.load_pem_x509_csr, backend
+        )
+        assert request.get_attribute_for_oid(
+            x509.oid.AttributeOID.CHALLENGE_PASSWORD
+        ) == b"challenge me!"
+
+    def test_invalid_attribute_for_oid(self, backend):
+        """
+        This test deliberately triggers an InternalError because to parse
+        CSR attributes we need to do a C cast. If we're wrong about the
+        type that would be Very Bad so this test confirms we properly explode
+        in the presence of the wrong types.
+        """
+        request = _load_cert(
+            os.path.join(
+                "x509", "requests", "challenge-invalid.der"
+            ), x509.load_der_x509_csr, backend
+        )
+        with pytest.raises(InternalError):
```

Comment:
I don't think we should have tests that expected InternalError, I think this should just be a ValueError or something